### PR TITLE
ATLAS-128: Support download of map images

### PR DESCRIPTION
**Pull request by HelioStrike**
JIRA issue: https://issues.openmrs.org/browse/ATLAS-128

I changed the resolution options from dimensions to scale (for now) because there don't seem to be any direct options for dimensions in html2canvas (there are width and height parameters, but they aren't working as expected). If we'd like dimensions, we'll probably need to scale the map canvas, take the screenshot, then scale it down again. I wasn't sure if the effort was worth it, as x2 generates a decent resolution image on a modern pc. What do you think?

Screenshot from 2019-07-02 18-04-20

### CODE REVIEW

**Comment by bmamlin** [comment]
Great work, @HelioStrike. I like having the work done on the client and, while we lose the ability to set specific dimensions, this approach has the advantage of generating an image based on the user's current settings. However, the "x1" and "x2" are not intuitive. I'd suggest getting rid of the dropdown for the Download button and, instead, just make the Download button generate the `scale: 2` image.

[canvas2image](https://github.com/hongru/canvas2image/) is a nice touch, but it's 5+ years old. It also generate a download file named "download" without an extension. Based on [this suggestion](https://github.com/hongru/canvas2image/issues/26), how about something like this instead:

```javascript
html2canvas(document.getElementById('map_canvas'), {
  scale: 2,
  useCORS: true,
  allowTaint:true,
}).then(function(canvas) {
  var link = document.createElement("a");
  link.download = "openmrs-atlas.png";
  link.href = canvas.toDataURL("image/png");
  link.click();
});
```

I think that would eliminate the need for `canvas2image` _and_ generate a friendlier filename.

**Comment by HelioStrike** [comment]
@bmamlin I made the changes. I had to append link to the document's body to make it work. This code works perfectly on mozilla, but says "Failed - Network error" on chrome for me. Canvas2Image doesn't work on chrome for me either, I didn't notice it until half an hour ago. Comments on stackoverflow say that the code works for them, so I suspect that this may be a problem in my computer.

File: views/index.ejs [hunk 1]
```diff
@@ -149,17 +150,8 @@
...
-       title="Click to download a screenshot" id ="down-screen">
-    <span><i class="fa fa-download"></i></span>Download
-  </div>
-  <div class = "dropDownOptionsDiv screen" id="screen">
-    <div class = "dropDownItemDiv screen" id="1024x768"><i class="fa fa-file-image-o"></i>1024x768</div>
-    <div class = "dropDownItemDiv screen" id="1280x1024"><i class="fa fa-file-image-o"></i>1920x1080</div>
-    <div class = "dropDownItemDiv screen" id="3840x2160"><i class="fa fa-file-image-o"></i>3840x2160</div>
-    <div class="separatorDiv"></div>
-  </div>
+<div class="atlas-container loginControl dropDownControl control" title="How to place my information on the Atlas" id ="download">
```
Comment (bmamlin) [inline views/index.ejs]: Are you sure the `loginControl` is supposed to be here? On a download button? If so, it is poorly named.

Comment (HelioStrike) [reply views/index.ejs]: Yup, I copied it from the other buttons. Removing it moves the button upwards.

Comment (bmamlin) [reply views/index.ejs]: Ok. Maybe we can do a CSS/UI cleanup separately.

**Pull request merged**